bugbug/similarity.py

```diff
@@ -195,6 +199,8 @@ def __init__(self, cleanup_urls=True, nltk_tokenizer=False):
             output_prefix="simdata.shdat", corpus=corpus_lsi, num_features=300
         )
 
+        joblib.dump(self, f"{self.__class__.__name__.lower()}.similarity")
```

Comment:
We should have this in common between all classes (so, the base class should do this).
You can either:
1) Add a `save` method to the base class which does this, and make the trainer script call the `save` function
2) Move the training code into a `train_model` function for each class, the base class will have a `train` function and a `train_model` abstract function. The `train` function will call `train_model` and then use joblib.dump. The trainer script will just call `train`.